Source/DynamicFeatureRegistry.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the MIT license. See LICENSE file in https://github.com/yahoo/Override for terms.
+
+import Foundation
+
+/// DynamicFeatureRegistry a subclass of FeatureRegistry which provides an interface for using
+/// a collection of dynamic features instead of just pre-defined ones. When a feature is added,
+/// there is an attempt to apply an override state from the underlying data FeatureStore.
+@objc open class DynamicFeatureRegistry: FeatureRegistry {
+
+    /// A local store of the current dynamic features
+    private var dynamicFeatures: [String : AnyFeature] = [:]
+
+    override var features: [LabeledItem] {
+        let dynamicLabeledItems = dynamicFeatures.values.reduce(into: []) { (result, feature) in
+            result.append(LabeledFeatureItem(label: feature.key, feature: feature))
+        }
+        let staticFeatures = super.features
+        /// merge both dynamic and static features
+        let allFeatures = dynamicLabeledItems + staticFeatures
+        return allFeatures
+    }
+
+    /// This method is used to configure and store a dynamic feature. It will attempt to
+    /// assign an `override` value from the FeatureStore if it was previously set and then stores it
+    /// locally for future use.
+    /// - Parameter feature: A new feature
+    /// - Returns: The configured feature
+    public func configureDynamicFeature(feature: AnyFeature) -> AnyFeature {
+        // Bootstrap the last saved override value, if any
+        feature.override = featureStore[feature.key]
+
+        // Store feature in `dynamicFeatures` if needed.
+        if dynamicFeatures[feature.key] == nil {
```

Comment:
If a similar feature instance was added previously, this method mutates the incoming feature and returns it without saving. That feels like a design flaw because it potentially breaks objects that it doesn't own and the caller can't tell if it was successful. I don't have a perfect solution to propose but I suugest replacing this check with a `guard` statement at the start of the method. If `feature.key` is already present in `dynamicFeatures`, it can take one of these paths:

 1. return that previous instance. (and document that)
 2. throw an exception.
 3. return nil.

Which one is appropriate depends on what you had in mind for the use case where multiple independent SDKs might be using the same registry instance.

Reply:
Really good points.

I think mutating and returning the same feature wasn't great. It was nice to consume but as you pointed out, the side effects were unclear. Additionally, this method didn't address what would happen if a dynamic feature conflicted with a statically defined one. 

I removed the return and added a simple getter method instead. It will be another line at the call site, assuming you want the enabled state of the feature immediately, but I think it's better overall.